server/src/main/java/org/apache/iotdb/db/mpp/aggregation/CountIfAccumulator.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.mpp.aggregation;
+
+import org.apache.iotdb.db.mpp.execution.operator.window.IWindow;
+import org.apache.iotdb.db.mpp.plan.expression.Expression;
+import org.apache.iotdb.db.mpp.plan.expression.binary.CompareBinaryExpression;
+import org.apache.iotdb.db.mpp.plan.expression.leaf.ConstantOperand;
+import org.apache.iotdb.tsfile.file.metadata.enums.TSDataType;
+import org.apache.iotdb.tsfile.file.metadata.statistics.Statistics;
+import org.apache.iotdb.tsfile.read.common.block.column.Column;
+import org.apache.iotdb.tsfile.read.common.block.column.ColumnBuilder;
+
+import java.util.List;
+import java.util.Map;
+
+public class CountIfAccumulator implements Accumulator {
+
+  // number of the point segment that satisfies the KEEP expression
+  private long countValue = 0;
+
+  // number of the continues data points satisfy IF expression
+  private long keep;
+
+  private final Evaluator keepEvaluator;
+
+  private final boolean ignoreNull;
+
+  private boolean lastPointIsSatisfy;
+
+  @FunctionalInterface
+  private interface Evaluator {
+    boolean evaluate();
+  }
```

Comment:
Maybe the code of Evaluator can be put in an indenpendent class?  So it can be directly use it in other codes when refer to keep parameter.

Reply:
> Maybe the code of Evaluator can be put in an indenpendent class? So it can be directly used in other codes when refer to `keep` parameter.

If there are more functions will use `keep`, it is a good idea to extract this code.